datafusion/optimizer/src/analyzer/type_coercion.rs

```diff
@@ -103,6 +103,16 @@ fn analyze_internal(
     // select t2.c2 from t1 where t1.c1 in (select t2.c1 from t2 where t2.c2=t1.c3)
     schema.merge(external_schema);
 
+    if let LogicalPlan::Filter(filter) = &mut plan {
+        if let Ok(new_predicate) = filter
+            .predicate
+            .clone()
```

Comment:
I found a way to avoid this clone: https://github.com/apache/datafusion/pull/11530